server/routers/admin.js

```diff
@@ -0,0 +1,213 @@
+          ],
+        },
+      },
+      { $unwind: { path: '$cartCount', preserveNullAndEmptyArrays: true } },
+      {
+        $group: {
+          _id: null,
+          totalCount: { $sum: '$cartCount.count' },
+          totalPrice: { $sum: '$price' },
+        },
+      },
+    ])
+      // return init value if there isn't carts and collection array is empty
+      .then((query) => query[0] ?? { totalCount: 0, totalPrice: 0 })
+  );
+}
+
+function getCarts() {
+  return Cart.aggregate([
+    {
+      $lookup: {
+        from: 'orders',
+        let: { orderIds: '$orderIds' },
+        as: 'lastUpdateAggregate',
+        pipeline: [
+          {
+            $match: {
+              $expr: { $in: ['$_id', '$$orderIds'] },
+            },
+          },
+          {
+            $group: {
+              _id: null,
+              lastOrderDate: { $max: '$time' },
+            },
+          },
+        ],
+      },
+    },
+    {
+      $lookup: {
+        from: 'users',
+        let: { userId: '$userId' },
+        as: 'user',
+        pipeline: [
+          {
+            $match: {
+              $expr: {
+                $eq: ['$$userId', '$_id'],
+              },
+            },
+          },
+        ],
+      },
+    },
+    { $unwind: { path: '$lastUpdateAggregate', preserveNullAndEmptyArrays: true } },
+    { $unwind: { path: '$user', preserveNullAndEmptyArrays: true } },
+    {
+      $addFields: {
+        lastUpdate: '$lastUpdateAggregate.lastOrderDate',
+        username: '$user.email',
+      },
+    },
+    { $project: { lastUpdateAggregate: 0, user: 0 } },
+  ]);
+}
+
+function getSingleCart(cartId) {
+  return Cart.aggregate([
+    {
+      $match: {
+        $expr: { $eq: ['$_id', ObjectId(cartId)] },
+      },
+    },
+    {
+      $lookup: {
+        from: 'orders',
+        let: { orderIds: '$orderIds' },
+        as: 'orders',
+        pipeline: [
+          {
+            $match: {
+              $expr: {
+                $and: [{ $in: ['$_id', '$$orderIds'] }],
+              },
+            },
+          },
+          { $sort: { time: -1 } },
+          {
+            $lookup: {
+              from: 'items',
+              localField: 'itemId',
+              foreignField: '_id',
+              as: 'item',
+            },
+          },
+          { $unwind: { path: '$item', preserveNullAndEmptyArrays: true } },
+        ],
+      },
+    },
+    {
+      $lookup: {
+        from: 'users',
+        let: { userId: '$userId' },
+        as: 'user',
+        pipeline: [
+          {
+            $match: {
+              $expr: {
+                $eq: ['$$userId', '$_id'],
+              },
+            },
+          },
+        ],
+      },
+    },
+    { $unwind: { path: '$user', preserveNullAndEmptyArrays: true } },
+    {
+      $addFields: {
+        username: '$user.email',
+      },
+    },
+  ]).then((query) => query[0]);
+}
+
+admin.get('/', (request, response) => {
+  const { token } = request.cookies;
```

Comment:
@IgorChaikin This lines are repeated at top of each route in `admin` probably you could reduce amount of repeated lines using `express middleware`, also you could read some information about `DRY principle`